doc/tntcxx_getting_started.rst

```diff
@@ -0,0 +1,524 @@
+   :start-after: doclabel06-2
+   :end-before: doclabel06-3
+   :language: c
+
+To reset connection after errors, that is, to clean up the error message and
+connection status, the ``Connection::reset()`` method is used.
+
+.. // TBD For the detailed information on the connector's API, refer to <link to the API document parts related to the API used here> -- implement this line when API document is ready.
+
+.. _gs_cxx_requests:
+
+Working with requests
+---------------------
+
+In this section, we will show how to:
+
+* :ref:`prepare different types of requests <gs_cxx_requests_prepare>`
+* :ref:`send the requests <gs_cxx_requests_send>`
+* :ref:`receive and handle responses <gs_cxx_requests_response>`.
+
+We will also go through the :ref:`case of having several connections <gs_cxx_requests_several>`
+and executing a number of requests from different connections simultaneously.
+
+In our example C++ application, we execute the following types of requests:
+
+* ping
+* replace
+* select.
+
+.. NOTE::
+
+   Examples on other request types, namely, ``insert``, ``delete``, ``upsert``,
+   and ``update``, will be added to this manual later.
+
+Each request method returns a request ID that is a sort of `future <https://en.wikipedia.org/wiki/Futures_and_promises>`_.
+It can be used to get the result of request execution once it is ready, that is,
+response. Requests are queued in the output buffer of connection
+until the ``Connector::wait()`` method is called.
+
+.. _gs_cxx_requests_prepare:
+
+Preparing requests
+~~~~~~~~~~~~~~~~~~~
+
+At this step, requests are encoded in the msgpack format and saved in the
+output connection buffer. They are ready to be sent but the network
+communication itself will be done later.
+
+Let's remind that for the requests manipulating with data we are dealing
+with the Tarantool space ``t`` :ref:`created earlier <gs_cxx_prereq_tnt_run>`,
+and the space has the following format:
+
+.. code-block:: lua
+
+    t:format({
+             {name = 'id', type = 'unsigned'},
+             {name = 'a', type = 'string'},
+             {name = 'b', type = 'number'}
+             })
+
+**ping**
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel07-1
+   :end-before: doclabel07-2
+   :language: c
+
+**replace**
+
+Equals to Lua request ``<space_name>:replace(pk_value, "111", 1)``.
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel08-1
+   :end-before: doclabel08-2
+   :language: c
+
+**select**
+
+Equals to Lua request ``<space_name>.index[0]:select({pk_value}, {limit = 1})``.
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel09-1
+   :end-before: doclabel09-2
+   :language: c
+
+.. _gs_cxx_requests_send:
+
+Sending requests
+~~~~~~~~~~~~~~~~~
+
+To send requests to the server side, we should invoke the ``client.wait()``
+method.
+
+.. code-block:: c
+
+   client.wait(conn, ping, WAIT_TIMEOUT);
+
+The ``wait()`` method takes the connection to poll (both IN and OUT),
```

Comment:
(both IN and OUT) -- I believe this clarification is better to be dropped (looks a bit misleading)

Reply:
Corrected.